packages/@winglang/wingc/src/type_check.rs

```diff
@@ -7496,27 +7496,58 @@ fn lookup_known_type(name: &'static str, env: &SymbolEnv) -> TypeRef {
 /// ```
 pub fn calculate_fqn_for_namespace(package_name: &str, package_root: &Utf8Path, path: &Utf8Path) -> String {
 	let normalized_root = normalize_path(&package_root, None);
-	let normalized = normalize_path(&path, None);
-	if normalized.starts_with("..") {
-		panic!(
-			"File path \"{}\" is not within the package root \"{}\"",
-			path, package_root
-		);
+	let normalized_path = normalize_path(&path, None);
+	if normalized_path.starts_with("..") {
+		report_diagnostic(Diagnostic {
+			message: format!(
+				"File path \"{}\" is not within the package root \"{}\"",
+				path, package_root
+			),
+			span: None,
+			annotations: vec![],
+			hints: vec![],
+			severity: DiagnosticSeverity::Error,
+		});
+		return package_name.to_string();
 	}
```

Comment:
Not sure if changing these panics into diagnostics is right per se, but it should result in fewer LSP crashes